core/src/mast/node/join_node.rs

```diff
@@ -49,8 +49,10 @@ impl JoinNode {
         Ok(Self { children, digest })
     }
 
-    #[cfg(test)]
-    pub fn new_test(children: [MastNodeId; 2], digest: RpoDigest) -> Self {
+    /// Returns a new [`JoinNode`] from values that are assumed to be correct.
+    /// Should only be used when the source of the inputs is trusted (e.g. deserialization).
+    pub fn new_unsafe(children: [MastNodeId; 2], digest: RpoDigest) -> Self {
+        // TODO(serge): debug_assert worthwhile? Need to pass in forest length to check children len
```

Comment:
I probably wouldn't change the signature just for `debug_assert` - so, maybe let's keep it as is.